test/test_utils/__init__.py

```diff
@@ -938,6 +995,20 @@ def get_processor_from_image_uri(image_uri):
     raise RuntimeError("Cannot find processor")
 
 
+def get_python_version_from_image_uri(image_uri):
```

Comment:
is this used anywhere in your code? seems like other functions like this follow a pattern of raising exception if no version is found

Reply:
Good call. It is better to return an Exception.